tensorflow_datasets/core/dataset_builder.py

```diff
@@ -383,33 +384,45 @@ def download_and_prepare(self, *, download_dir=None, download_config=None):
           # DatasetInfo.read_from_directory to possibly restore these attributes
           # when reading from package data.
 
-          # Skip statistics computation if tfdv isn't present
-          try:
-            import tensorflow_data_validation  # pylint: disable=g-import-not-at-top,import-outside-toplevel,unused-import  # pytype: disable=import-error
-            skip_stats_computation = False
-          except ImportError:
-            skip_stats_computation = True
-
-          splits = list(self.info.splits.values())
-          statistics_already_computed = bool(
-              splits and splits[0].statistics.num_examples)
-          # Update DatasetInfo metadata by computing statistics from the data.
-          if (skip_stats_computation or
-              download_config.compute_stats == download.ComputeStatsMode.SKIP or
-              download_config.compute_stats == download.ComputeStatsMode.AUTO
-              and statistics_already_computed
-             ):
-            logging.info(
-                "Skipping computing stats for mode %s.",
-                download_config.compute_stats)
-          else:  # Mode is forced or stats do not exists yet
-            logging.info("Computing statistics.")
-            self.info.compute_dynamic_properties()
           self.info.download_size = dl_manager.downloaded_size
-          # Write DatasetInfo to disk, even if we haven't computed statistics.
           self.info.write_to_directory(self._data_dir)
     self._log_download_done()
 
+  def generate_statistics(self, compute_stats=None):
+    """Generate statistics for Dataset
+
+    Args:
+      compute_stats: `tfds.download.ComputeStats`, whether to compute
+        statistics over the generated data. Defaults to `AUTO`.
+
+    Note: Requires `tensorflow_data_validation`.
+    """
+    compute_stats = download.ComputeStatsMode(compute_stats or
+                                              download.ComputeStatsMode.AUTO)
+
+    splits = list(self.info.splits.values())
+    statistics_already_computed = bool(
+        splits and splits[0].statistics.num_examples)
```

Comment:
To check whether statistics are already computed, we should check for the presence or not of statistics.json